src/docs/patterns/icons.mdx

```diff
@@ -0,0 +1,172 @@
+
+-   [Show an example of a hero banner, with different statuses. If the count is 0, make it inactive; if it is a positive number, make it active]
+
+## Progress Icons
+
+Progress icons are used as a mini data visualization to display percentages over a whole. It can be the remaining device's health, battery life percentage, etc.
+
+-   [Show examples of percentage icons used in a real application]
+
+Brightlayer currently offers the following progress icons to its designers and developers:
+
+<ProgressIconCard />
+
+## Internationalization
+
+In right-to-left languages, icons with directionalities must be mirrored. Read more about it on our [internationalization pattern page](</patterns/internationalization#bidirectionality--right-to-left-(rtl)-support>).
+
+# Design your own icons
+
+Brightlayer UI welcomes new suggestions to our existing icon repositories. However, before you start designing, there are a few things that you shall keep in mind.
+
+## No Single-Purpose Icons
+
+If you find yourself struggling for more than 30 seconds to design an icon, your users are likely going to be perplexed.
+
+Icons designed for a very specific purpose, such as “transfer admin rights” instead of “transfer”, tend to be so complicated that the icon no longer serves as an easy-to-recognize symbol, because it is impractical to pack so much information in a confined 24x24 space.
+
+Similarly, avoid inventing an icon for a very specific device model, unless this device stands out and is iconic in the industry (example: Dyson fans). If an average user cannot easily tell devices apart from each other in real life from a glance, a 24x24 dp icon usually contains fewer details and more abstractions, and will not bring a positive user experience.
+
+## Existing Icon Repositories
+
+Before you are committed to your artwork, make sure to check existing icon resources. In addition to Material Design's official icon set and BLUI icon set, materialdesignicons.com is a community-driven icon repository that follows Material Design's icon guidelines.
+
+You are welcome to take design inspirations from other sources like Font Awesome or follow regulation documents such as those from ANSI and ISO. You are also encouraged to refer to your competitors and see if they have any established conventions. However, you should follow the style guide, as explained below.
+
+## Follow the Style Guide
+
+<MaterialDesignDescription
+    title={'System icons'}
+    url={'https://material.io/design/iconography/system-icons.html#'}
+    description={`Material Design's guidelines on creating your own icons.`}
+/>
+
+You must follow the existing Material Design style guide when creating your own icons. You may get inspired from other sources, but you are required to alter the style so that they fit with existing BLUI icons.
```

Comment:
```suggestion
You must follow the existing Material Design style guide when creating your own icons. You may take inspiration from other sources, but you must update the style so that they fit with existing BLUI icons.
```